### PR TITLE
[CROSSDATA-298] Stop streaming process

### PR DESCRIPTION
**Pull request by darroyocazorla**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/crossdata/models/EphemeralTableModel.scala [hunk 1]
```diff
@@ -18,7 +18,7 @@ package org.apache.spark.sql.crossdata.models
 import org.apache.spark.sql.types.StructType
 
 case class EphemeralTableModel(name: String,
-                               schema: StructType,
+                               schema: Option[StructType],
```
Comment (compae) [inline core/src/main/scala/org/apache/spark/sql/crossdata/models/EphemeralTableModel.scala]: Change to the last position in argument list, then we can create EphemeralTableModel(name, options)

File: streaming/src/main/scala/com/stratio/crossdata/streaming/actors/EphemeralStatusActor.scala [hunk 2]
```diff
@@ -20,31 +20,69 @@ import akka.actor.Actor
...
+    case AddListener if !listenerAdded =>
       doAddListener()
       context.become(receive(true))
   }
 
+  // TODO require tableModel
+  private val ephemeralTableModel = ephemeralTMDao.dao.get(ephemeralTableName).get
+  private val delayMs = ephemeralTableModel.options.atomicWindow * 1000
+
+  context.system.scheduler.schedule(delayMs milliseconds, delayMs milliseconds, self, CheckStatus)(context.dispatcher)
```
Comment (compae) [inline streaming/src/main/scala/com/stratio/crossdata/streaming/actors/EphemeralStatusActor.scala]: Could you make this inside override preStart?

File: streaming/src/main/scala/com/stratio/crossdata/streaming/actors/EphemeralStatusActor.scala [hunk 3]
```diff
@@ -20,31 +20,69 @@ import akka.actor.Actor
...
   override def receive: Receive = receive(listenerAdded = false)
 
   def receive(listenerAdded: Boolean): Receive = {
+    case CheckStatus => doCheckStatus()
     case GetStatus => doGetStatus()
     case SetStatus(status) => doSetStatus(status)
-    case AddListener if !listenerAdded=>
+    case AddListener if !listenerAdded =>
       doAddListener()
       context.become(receive(true))
```
Comment (compae) [inline streaming/src/main/scala/com/stratio/crossdata/streaming/actors/EphemeralStatusActor.scala]: Nice! I like this!!

Comment (compae) [inline streaming/src/main/scala/com/stratio/crossdata/streaming/actors/EphemeralStatusActor.scala]: Nice! I like this!!

File: streaming/src/main/scala/com/stratio/crossdata/streaming/helpers/CrossdataStatusHelper.scala [hunk 4]
```diff
@@ -60,61 +77,20 @@ object CrossdataStatusHelper extends SparkLoggerComponent {
...
-      queryActor <- ephemeralQueryActor
-      statusActor <- ephemeralStatusActor
-    } yield {
-      statusActor ! SetStatus(EphemeralExecutionStatus.Stopped)
-      actorSystem.stop(queryActor)
-      actorSystem.stop(statusActor)
-    }
+
+    ephemeralQueryActor.foreach(_ ! PoisonPill)
+    ephemeralStatusActor.foreach(_ ! PoisonPill)
```
Comment (compae) [inline streaming/src/main/scala/com/stratio/crossdata/streaming/helpers/CrossdataStatusHelper.scala]: do you need set the status Stopped?

statusActor ! SetStatus(EphemeralExecutionStatus.Stopped)

File: streaming/src/main/scala/com/stratio/crossdata/streaming/helpers/CrossdataStreamingHelper.scala [hunk 5]
```diff
@@ -76,18 +75,19 @@ object CrossdataStreamingHelper extends SparkLoggerComponent {
...
     val kafkaOptionsMerged = kafkaOptions.copy(
       partitionOutput = ephemeralQuery.options.get(PartitionKey).orElse(kafkaOptions.partitionOutput),
       additionalOptions = ephemeralQuery.options)
     val xdContext = XDContext.getOrCreate(rdd.context, parseZookeeperCatalogConfig(zookeeperConf))
-    // TODO add optional schema => xdContext.read.schema; at least -> sampling ratio
-    val df = xdContext.read.json(filterRddWithWindow(rdd, ephemeralQuery.window))
+    // TODO add sampling ratio
+    val dfReader = xdContext.read
+    val dfReaderWithschema = ephemeralTable.schema.map(dfReader.schema).getOrElse(dfReader)
+    val df = dfReaderWithschema.json(filterRddWithWindow(rdd, ephemeralQuery.window))
```
Comment (compae) [inline streaming/src/main/scala/com/stratio/crossdata/streaming/helpers/CrossdataStreamingHelper.scala]: YES

File: core/src/main/scala/com/stratio/crossdata/launcher/SparkJobLauncher.scala [hunk 6]
```diff
@@ -25,113 +25,61 @@ import org.apache.spark.sql.crossdata.config.StreamingConstants
...
-        }
-    }(executionContext)
-
-    launchedProcess.onFailure{
-      case throwable =>
-        logError(throwable.getMessage)
-    }(executionContext)
-
+    sparkConf.map({ case (key, value) => sparkLauncher.setConf(key, value) })
+    new SparkJob(sparkLauncher)(executionContext)
```
Comment (compae) [inline core/src/main/scala/com/stratio/crossdata/launcher/SparkJobLauncher.scala]: Good abstraction!

File: pom.xml [hunk 7]
```diff
@@ -108,8 +108,8 @@
         <mariadb.version>1.2.3</mariadb.version>
         <postgresql.version>9.2-1003-jdbc4</postgresql.version>
         <derby.version>10.12.1.1</derby.version>
-        <jackson4s.version>3.2.11</jackson4s.version>
-        <common.utils.version>0.4.3</common.utils.version>
+        <jackson4s.version>3.2.10</jackson4s.version>
+        <common.utils.version>0.5.0-SNAPSHOT</common.utils.version>
```
Comment (compae) [inline pom.xml]: Tomorrow I generate one release and we can change to the new version

**Comment by compae** [comment]
Squash in one commit?

**Comment by darroyocazorla** [comment]
Thanks! changes made

**Pull request merged**